File: src/spoj/PRIME1.java
package spoj;

        import java.io.BufferedReader;
        import java.io.IOException;
        import java.io.InputStreamReader;
        import java.util.ArrayList;

public class PRIME1 {
    static private ArrayList<Integer> primes = new ArrayList<>();

    private static void computePrimes() {
        for (int i = 2; i <= Math.sqrt(1000000000); i++) {
            if (isPrime(i)) primes.add(i);
        }
    }

    private static boolean isPrime(int n) {
        boolean isPrime = true;

        for (int j = 0; j < primes.size() && primes.get(j) <= Math.sqrt(n); j++) {
            if (n % primes.get(j) == 0) {
                isPrime = false;
                break;
            }
        }

        return isPrime;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        computePrimes();
        int t = Integer.parseInt(in.readLine());

        while (t-- > 0) {
            String[] s = in.readLine().split(" ");
            int m = Integer.parseInt(s[0]);
            int n = Integer.parseInt(s[1]);

            for (int i = Math.max(m, 2); i <= n; i++) {
                if (isPrime(i)) System.out.println(i);
            }
            System.out.println();
        }
    }
}

File: cpp/spoj/prime1.cc
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

vector<int> primes;

bool is_prime(int n) {
    if (n <= 1) {
        return false;
    }
    for (int i = 0;  i < primes.size() && primes[i] * primes[i] <= n; i++) {
        if (n % primes[i] == 0) {
            return false;
        }
    }
    return true;
}

void populate_primes(int till) {
    primes.push_back(2);
    for (int i = 3; i <= sqrt(till); i += 2) {
        if (is_prime(i)) {
            primes.push_back(i);
        }
    }
}

int main() {
    int t;
    cin >> t;

    populate_primes(1000000000);
    while (t-- > 0) {
        int n, m;
        cin >> n >> m;
        for (int i = n; i <= m; i++) {
            if (is_prime(i)) {
                cout << i << "\n";
            }
        }
        cout << "\n";
    }
    return 0;
}
